cupy/linalg/_util.py

```diff
@@ -19,17 +19,26 @@ def _assert_cupy_array(*arrays):
                 'cupy.linalg only supports cupy.ndarray')
 
 
-def _assert_rank2(*arrays):
+def _assert_2d(*arrays):
     for a in arrays:
         if a.ndim != 2:
             raise linalg.LinAlgError(
                 '{}-dimensional array given. Array must be '
                 'two-dimensional'.format(a.ndim))
 
 
-def _assert_nd_squareness(*arrays):
+def _assert_stacked_2d(*arrays):
     for a in arrays:
-        if max(a.shape[-2:]) != min(a.shape[-2:]):
+        if a.ndim != 2:
+            raise linalg.LinAlgError(
+                '{}-dimensional array given. Array must be '
+                'at least two-dimensional'.format(a.ndim))
```

Comment:
`_assert_stacked_2d()` doesn't make sense to me? It rules out the possibility of having a stack of 2D matrices.

Reply:
Oops 😳